lms/djangoapps/learner_recommendations/views.py

```diff
@@ -105,5 +107,8 @@ def get(self, request, course_id):
         )
         recommended_courses = map(course_data_for_discovery_card, filtered_courses)
         recommended_courses = recommended_courses[:recommendation_count]
-
-        return Response({"is_control": is_control, "courses": recommended_courses}, status=200)
+        if not is_enterprise_learner(request.user):
+            program_upsell = get_programs_based_on_course(course_key, user_country_code, request.user)
```

Comment:
The enterprise learner check should be below the waffle flag check. Both recommendations should not be shown to enterprise learners. Please review technical approach document.